presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryMetadata.java

```diff
@@ -29,6 +29,7 @@
 
     private final String query;
     private final String queryHash;
+    private final Optional<String> queryTemplateHash;
```

Comment:
Does it need to be optional?

Reply:
Yes, `QueryMetadata` is constructed from either `BasicQueryInfo` or `QueryInfo`. To keep `BasicQueryInfo` lightweight -- used for immediate failed queries and created queries, we do not store the query template hash in `BasicQueryInfo`. For this reason, we allow this to be optional when constructing `QueryMetadata` with `BasicQueryInfo`.